### PR TITLE
Bug Fix - stop crashing when copying a tar with an ImageIndex

### PR DESCRIPTION
**Pull request by DennisDenuto**
Opening as a draft as there is still work to do. Namely

- [x] backfill test copying a tarfile with an ImageIndex

### CODE REVIEW

File: pkg/imgpkg/cmd/copy.go [hunk 1]
```diff
@@ -242,6 +242,10 @@ func (o *CopyOptions) writeBundleLockOutput(bundle *bundle.Bundle) error {
 func processedImagesMediaType(processedImages *ctlimgset.ProcessedImages) []string {
 	everyMediaType := []string{}
 	for _, image := range processedImages.All() {
+		if image.Image == nil {
```
Comment (cppforlife) [inline pkg/imgpkg/cmd/copy.go]: dont we need to recurse into image index as well?

Comment (DennisDenuto) [reply pkg/imgpkg/cmd/copy.go]: We do! we discovered this while TDD'ing. See updated impl.

File: pkg/imgpkg/cmd/copy.go [hunk 2]
```diff
@@ -242,17 +243,42 @@ func (o *CopyOptions) writeBundleLockOutput(bundle *bundle.Bundle) error {
...
-		for _, layer := range layers {
-			mediaType, err := layer.MediaType()
+		if image.ImageIndex != nil {
+			indexManifest, err := image.ImageIndex.IndexManifest()
 			if err != nil {
 				continue
 			}
-			everyMediaType = append(everyMediaType, string(mediaType))
+			for _, descriptor := range indexManifest.Manifests {
+				image, err := image.ImageIndex.Image(descriptor.Digest)
```
Comment (cppforlife) [inline pkg/imgpkg/cmd/copy.go]: i might be wrong, i thought that indexes could contain other indexes as well.

Comment (DennisDenuto) [reply pkg/imgpkg/cmd/copy.go]: Added logic to recurse through image indexes

**Review by joaopapereira — APPROVED** [review]
LGTM

**Pull request merged**